test/tests/functional/pbs_test_tpp.py

```diff
@@ -825,10 +920,162 @@ def test_COMM_ROUTERS_with_nondefault_pbs_conf(self):
                           job=True, interactive=True, resv=True,
                           resv_job=True)
 
+    def calculate_no_of_wait_threads(self, comm_pid):
+        """
+        This function calculate the number of wait threads
+        :param comm_pid: Comm's Pid
+        :type comm_pid: Integer.
+        """
+        ps_cmd = "ps -eT | grep pbs_comm | wc -l"
+        rc = self.server.du.run_cmd(self.server.hostname,
+                                    cmd=ps_cmd, as_script=True)
+        num_wait_threads = int(rc['out'][0]) - 1
+        return num_wait_threads
+
+    def test_PBS_COMM_THREADS(self):
+        """
+        Test allowable values for PBS_COMM_THREADS
+        """
+        threads = [1, 2, 100, 101, "T"]
+        for wait_thread in threads:
+            a = {'PBS_COMM_THREADS': wait_thread}
+            self.set_pbs_conf(host_name=self.mom.shortname, conf_param=a)
+            comm_pid = self.comm.get_pid()
+            num_wait_thread = self.calculate_no_of_wait_threads(comm_pid)
+            if wait_thread == 1 or wait_thread == 101:
+                wait_thread = -1
```

Comment:
Use a different variable, since wait_thread is the loop variable, it is not suggested to change it directly. Same for line number 948. Maybe num_threads.